glide-core/redis-rs/redis/src/commands/cluster_scan.rs

```diff
@@ -130,11 +146,18 @@ impl ScanStateRC {
     }
 }
 
-/// This trait defines the methods for interacting with a Redis cluster during scanning.
-#[async_trait]
+/// Trait defining methods for interacting with a cluster during scanning.
+///
+/// Implementors of this trait provide functionalities such as retrieving the address
+/// for a slot, getting the epoch of an address, routing commands, and checking cluster coverage.
 pub(crate) trait ClusterInScan {
     /// Retrieves the address associated with a given slot in the cluster.
-    async fn get_address_by_slot(&self, slot: u16) -> RedisResult<Arc<String>>;
+    async fn get_address_by_slot(
+        &self,
+        slot: u16,
+        scanned_slots_map: &mut SlotsBitsArray,
+        allow_non_covered_slots: bool,
```

Comment:
why not pass here the complete args structure? why only this field?

Reply:
Removed this function with all the trait removal, but just for the idea itself.
Args is something else, is what the user interface uses, and the sender send to the internal.
But anyway, why pass more than needed? Why pay more than required?